Support the Community Highlander status hooks (#1)

pledbrook opened the pull request:
Adds a check for whether the Community Highlander is active and meets the minimum version of 1.19. If that's the case, this change makes the soldier list items use the highlander's `GetPersonnelStatusSeparate()` function for rendering soldiers' statuses.

This makes the mod compatible with any other mod that uses those hooks to add or modify soldier statuses.

Closes #1.

pledbrook committed:
Support the Community Highlander status hooks (#1)

Adds a check for whether the Community Highlander is active and meets
the minimum version of 1.19. If that's the case, this change makes the
soldier list items use the highlander's `GetPersonnelStatusSeparate()`
function for rendering soldiers' statuses.

This makes the mod compatible with any other mod that uses those hooks
to add or modify soldier statuses.

Closes #1.

pledbrook committed:
Fix Days -> Hours for CHL integration (#1)

The Community Highlander doesn't automatically convert days to hours in
status strings, so this adds an event listener for the
`OverridePersonnelStatusTime` event to perform that conversion. This
ensures that the behaviour with and without the CHL is identical.

Note that this change means the mod must be compiled against the
Community Highlander.

pledbrook commented:
I recommend a squash and merge in this case, since I simply didn't realise the listener needed to be included in the mod (LWOTC has its own listener to do the conversion).

Note that the support for Days -> Hours conversion requires the Community Highlander classes to be on the compilation classpath when building this mod, as described in the [CHL README](https://github.com/X2CommunityCore/X2WOTCCommunityHighlander/blob/master/README.md#building-against-the-highlander).

Xymanek commented on DetailedSoldierListWOTC/Src/DetailedSoldierListWOTC/Classes/X2EventListener_DetailedSoldierList.uc:
```diff
@@ -0,0 +1,73 @@
+
+static function CHEventListenerTemplate CreateStatusListeners()
+{
+	local CHEventListenerTemplate Template;
```
This will crash if CHL isn't loaded. I suggest adding a check before `CreateStatusListeners()` call

pledbrook committed:
Protect listener against missing CHL

`X2EventListener_DetailedSoldierList` makes use of Community Highlander
classes, so I now skip the template creation if the appropriate CHL
isn't loaded. This is a defensive measure in case it causes crashes,
although the game seems to handle the missing CHL fine in simple
testing.

bountygiver commented:
moved to #5
